src/groupeddataframe/groupeddataframe.jl

```diff
@@ -508,6 +522,14 @@ function Base.haskey(gd::GroupedDataFrame, key::NamedTuple{N}) where {N}
     return haskey(gd, Tuple(key))
 end
 
+function Base.haskey(gd::GroupedDataFrame, key::Dict{T, S}) where {T<:Union{Symbol, <:AbstractString}, S}
+    if length(key) != length(gd.cols) || any(Symbol(n) ∉ _names(gd) for n in keys(key))
```

Comment:
As commented above - if we move this code to `_dict_to_tuple` then we will not have code duplication.

Reply:
I am unresolving this because this check is incorrect. Also if you move the `if` in `to_index` to `_dict_to_tuple` it will not be needed at all.

Reply:
this can be resolved, sorry.